Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchyBasedLinkResolver.swift

```diff
@@ -58,6 +57,42 @@ final class PathHierarchyBasedLinkResolver {
         }
     }
     
+    /// Returns the direct descendants of the given page that match the given source language filter.
+    ///
+    /// A descendant is included if it has a language representation in at least one of the languages in the given language filter or if the language filter is empty.
+    ///
+    /// - Parameters:
+    ///   - reference: The identifier of the page whose descendants to return.
+    ///   - languagesFilter: A set of source languages to filter descendants against.
+    /// - Returns: The references of each direct descendant that has a language representation in at least one of the given languages.
+    func directDescendants(of reference: ResolvedTopicReference, languagesFilter: Set<SourceLanguage>) -> Set<ResolvedTopicReference> {
+        guard let id = resolvedReferenceMap[reference] else { return [] }
+        let node = pathHierarchy.lookup[id]!
+        
+        func directDescendants(of node: PathHierarchy.Node) -> [ResolvedTopicReference] {
+            return node.children.flatMap { _, container in
+                container.storage.compactMap { element in
+                    guard let childID = element.node.identifier, // Don't include sparse nodes
+                          !element.node.specialBehaviors.contains(.excludeFromAutomaticCuration),
+                          languagesFilter.isEmpty || !element.node.languages.isDisjoint(with: languagesFilter)
+                    else {
+                        return nil
+                    }
+                    return resolvedReferenceMap[childID]
+                }
+            }
+        }
+        
+        var results = Set<ResolvedTopicReference>()
+        if node.languages.isSuperset(of: languagesFilter) {
+            results.formUnion(directDescendants(of: node))
+        }
+        if let counterpart = node.counterpart, counterpart.languages.isSuperset(of: languagesFilter) {
+            results.formUnion(directDescendants(of: counterpart))
+        }
```

Comment:
I'm still not sure why we're checking for supersets here. What happens when a node only exists in Swift, but the language filter is set to both Swift and Objective-C? My understanding of the `isSuperset` check here is that it would skip over this node and not collect its children, since it doesn't have a representation in all the requested languages.

If the goal is to include nodes "in at least one of the given languages", shouldn't we be checking for `!isDisjoint` here as well?

Reply:
I had written it like this to support empty language filters before updating the code to check `languagesFilter.isEmpty || !self.languages.isDisjoint(with: languagesFilter)`. In practice there's only one place that calls this and it only passes a filter with one language. 

I updated the 3 places in this function to check if a node matches a language filter using the same logic in https://github.com/apple/swift-docc/pull/854/commits/23450ae955b816a5f0edf7bad6f7a339eb476376